### DISCUSSION TITLE
errors migrating from 3.3.6 to 3.5.0 - TypeError: __init__() missing 1 required positional argument: 'model'

### DISCUSSION THREAD

**Discussion by sivasudheer6** [netbox-community/netbox#14426] [Help Wanted!] [answered]
File "/opt/netbox-3.5.0/netbox/scripts/examples.py", line 19, in NewBranchScript
    switch_model = ObjectVar(
TypeError: __init__() missing 1 required positional argument: 'model'

**Comment by candlerb** [comment, marked as answer]
I think the error message is clear: there is a script called "examples.py" in the scripts directory, and that script is not valid with the new version of Netbox.  I think you must have copied the one from [here](https://docs.netbox.dev/en/stable/customization/custom-scripts/) but it seems to be broken.

As a quick fix, you can move the file out of the way and replace it with a zero-byte file:

```
cd /opt/netbox-3.5.0/netbox/scripts
mv examples.py examples.py.broken
touch examples.py
```

**Comment by sivasudheer6** [comment]
Hey Candlerb, Thank you for your insights, I'm able to resolve that error but encountering a new issue:
ModuleNotFoundError: No module named 'graphiql_debug_toolbar'
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
root@dxlvmdcops01:/opt/netbox-3.6.6# [2023-12-05 16:25:09 +0000] [74454] [ERROR] Exception in worker process

**Reply by candlerb** [reply]
Try the following command:

```
/opt/netbox/venv/bin/pip list | grep -i toolb
```

I get:

```
django-debug-toolbar          4.2.0
django-graphiql-debug-toolbar 0.2.0
```

Also, try this:

```
grep ExecStart /etc/systemd/system/netbox.service
```

I get:

```
ExecStart=/opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi
```

Finally, you appear to be using the tarball method of installation, which requires a symlink for `/opt/netbox`. What does this command show?

```
ls -l /opt/netbox
```

Your first posting claims to be running netbox 3.5.0, your second one claims to be 3.6.6.  I suspect you made an incomplete upgrade.

**Reply by candlerb** [reply]
Your other post in #14435 makes it clear that you're running the wrong instance of gunicorn, presumably meaning that your netbox.service is wrong. That's why the worker can't find the packages installed into the virtualenv.

**Comment by sivasudheer6** [comment]
cd /opt/netbox
. venv/bin/activate
I performed this before starting the service and the issue is resolved now

**Reply by candlerb** [reply]
Then you're not running Netbox in the official way, which would have been helpful to know before you started this thread, but I'm glad it's working for you now.
